apps/fdsnws/share/js/event-time-selection.js

```diff
@@ -0,0 +1,160 @@
+}
+
+const debouncedEventLookup = debounce(async function(eventId) {
+  const statusDiv = document.getElementById('event-status');
+  
+  if (!eventId.trim()) {
+    currentOriginTime = null;
+    statusDiv.textContent = '';
+    statusDiv.className = '';
+    return;
+  }
+
+  if (eventId.length < 3) {
+    statusDiv.textContent = 'Enter complete Event ID...';
+    statusDiv.className = 'status loading';
+    return;
+  }
+
+  statusDiv.textContent = 'Fetching event...';
+  statusDiv.className = 'status loading';
+
+  try {
+    currentOriginTime = await fetchEventOriginTime(eventId);
+    statusDiv.textContent = `Origin: ${currentOriginTime.toISOString().replace(/\.\d{3}Z$/, '')}`;
+    statusDiv.className = 'status success';
+    updateTimeFields();
+  } catch (error) {
+    statusDiv.textContent = error.message;
+    statusDiv.className = 'status error';
+    currentOriginTime = null;
+  }
+}, 800);
+
+function clearEventData() {
+  document.getElementById('event-id').value = '';
+  document.getElementById('before-time').value = '';
+  document.getElementById('after-time').value = '';
+  document.querySelector('input[name="starttime"]').value = '';
+  document.querySelector('input[name="endtime"]').value = '';
+  document.getElementById('event-status').textContent = '';
+  document.getElementById('event-status').className = '';
+  currentOriginTime = null;
+  
+  const form = document.getElementById('query-form');
+  form.dispatchEvent(new Event('input', { bubbles: true }));
+}
+
+async function initializeEventTimeSelection() {
+  // Check event service on load and configure UI
+  const serviceAvailable = await checkEventService();
+  const eventSection = document.querySelector('.event-section');
+  const eventLegend = eventSection.querySelector('legend');
+  
+  if (serviceAvailable) {
+    eventLegend.textContent = 'Event-based Time Selection ✓';
+    eventLegend.style.color = '#28a745';
+  } else {
+    eventLegend.textContent = 'Event-based Time Selection (Event service unavailable)';
+    eventLegend.style.color = '#dc3545';
+    eventSection.disabled = true;
+    
+    // Disable all inputs in the section
+    const inputs = eventSection.querySelectorAll('input, button');
+    inputs.forEach(input => input.disabled = true);
+  }
+  
+  // Only set up event listeners if service is available
+  if (serviceAvailable) {
+    document.getElementById('event-id').addEventListener('input', (e) => {
+      debouncedEventLookup(e.target.value);
+    });
+    
+    document.getElementById('before-time').addEventListener('input', updateTimeFields);
+    document.getElementById('after-time').addEventListener('input', updateTimeFields);
+    document.getElementById('clear-event').addEventListener('click', clearEventData);
+  }
+}
+
+// Initialize when DOM is loaded
+if (document.readyState === 'loading') {
+  document.addEventListener('DOMContentLoaded', initializeEventTimeSelection);
+} else {
+  initializeEventTimeSelection();
+}
```

Comment:
This is overcomplicated - drop the if, we just need the one line `document.addEventListener('load', initializeEventTimeSelection);`. Additionally, I would move this line back to the inline script in dataselect-builder.html to be more in line with the existing approach used for fdsnwsInitQueryForm.